commands/displayers/output.go

```diff
@@ -71,97 +31,51 @@ type Displayable interface {
 	JSON(io.Writer) error
 }
 
+// Displayer has the display options, the item to display, and where to display to
 type Displayer struct {
-	NS     string
-	Config doctl.Config
-	Item   Displayable
-	Out    io.Writer
+	OutputType string
+	ColumnList string
+	NoHeaders  bool
+
+	Item Displayable
+	Out  io.Writer
 }
 
+// Display ends up rendering the content in one of two formats (text|json)
 func (d *Displayer) Display() error {
-	output, err := d.Config.GetString(doctl.NSRoot, "output")
-	if err != nil {
-		return nil
-	}
-
-	if output == "" {
-		output = "text"
-	}
-
-	switch output {
+	switch d.OutputType {
 	case "json":
 		if containsOnlyNilSlice(d.Item) {
 			_, err := d.Out.Write([]byte("[]"))
 			return err
 		}
 		return d.Item.JSON(d.Out)
 	case "text":
-		cols, err := handleColumns(d.NS, d.Config)
-		if err != nil {
-			return err
+		var cols []string
+		for _, c := range strings.Split(strings.Join(strings.Fields(d.ColumnList), ""), ",") {
+			if c != "" {
+				cols = append(cols, c)
+			}
 		}
 
-		return displayText(d.Item, d.Out, cols)
+		return DisplayText(d.Item, d.Out, d.NoHeaders, cols)
 	default:
 		return fmt.Errorf("unknown output type")
 	}
 }
 
-func writeJSON(item interface{}, w io.Writer) error {
-	b, err := json.Marshal(item)
-	if err != nil {
-		return err
-	}
-
-	var out bytes.Buffer
-	err = json.Indent(&out, b, "", "  ")
-	if err != nil {
-		return err
-	}
-	_, err = out.WriteTo(w)
-
-	return err
-}
-
-// containsOnlyNiSlice returns true if the given interface's concrete type is
-// a pointer to a struct that contains a single nil slice field.
-func containsOnlyNilSlice(i interface{}) bool {
-	if reflect.TypeOf(i).Kind() != reflect.Ptr {
-		return false
-	}
-
-	element := reflect.ValueOf(i).Elem()
-	if element.NumField() != 1 {
-		return false
-	}
-
-	slice := element.Field(0)
-	if slice.Kind() != reflect.Slice {
-		return false
-	}
-
-	if slice.Cap() != 0 {
-		return false
-	}
-	if slice.Len() != 0 {
-		return false
-	}
-	if slice.Pointer() != 0 {
-		return false
-	}
-
-	return true
-}
-
-func displayText(item Displayable, out io.Writer, includeCols []string) error {
-	w := newTabWriter(out)
+// DisplayText ends up writing tabbed content to the passed in io.Writer
+// while potentially adding our removing headers.
```

Comment:
`our` should be `or`

Reply:
I guess if I want to be ultimate strict about this, the comment should read in the form,

```go
// DisplayText writes tabbed content...
```